src/main/java/org/candlepin/subscriptions/tally/InventoryAccountUsageCollector.java

```diff
@@ -307,17 +307,23 @@ private Host handleDuplicateHost(Set<String> duplicateInstanceIds, Host host1, H
 
   public static void populateHostFieldsFromHbi(
       Host host, InventoryHostFacts inventoryHostFacts, NormalizedFacts normalizedFacts) {
-    if (inventoryHostFacts.getInventoryId() != null) {
-      host.setInventoryId(inventoryHostFacts.getInventoryId().toString());
-      // We assume that the instance ID for any given HBI host record is the inventory ID; compare
-      // to an OpenShift Cluster from Prometheus data, where we use the cluster ID.
-      if (host.getInstanceId() == null) {
-        // Don't overwrite the instanceId if already set, since that would cause potential
-        // duplicates in the serviceInstances map in AccountServiceInventory.
+    // Don't overwrite the instanceId if already set, since that would cause potential
+    // duplicates in the serviceInstances map in AccountServiceInventory.
+    if (host.getInstanceId() == null) {
```

Comment:
@kahowell don't we want to overwrite the instance ID even when it already sets? Asking because we would not update the existing data that used to use the inventory ID.

Reply:
Actually, yes, we should in the case our algorithm has selected a more accurate id.

Reply:
Changed